courses/management/commands/create_enrollment.py

```diff
@@ -84,9 +84,24 @@ def handle(self, *args, **options):
                     )
                 }
             )
+        # Fetch the latest product version.
+        product_version = latest_product_version(product)
+
+        # Calculate the total paid price after applying coupon.
+        total_price_paid = get_product_version_price_with_discount(
+            coupon_version=coupon_version, product_version=product_version
+        )
+
+        # Create an order.
+        order = Order.objects.create(
+            status=Order.FULFILLED, purchaser=user, total_price_paid=total_price_paid
+        )
 
         successful_enrollments, edx_request_success = create_run_enrollments(
-            user, [run], keep_failed_enrollments=options["keep_failed_enrollments"]
+            user,
+            [run],
+            keep_failed_enrollments=options["keep_failed_enrollments"],
+            order=order,
```

Comment:
@arslanashraf7 We have an entry for the order in model `Order` even if the enrollment failed here due to any reason.